### DISCUSSION TITLE
Avoid Jit compilation in each process

### DISCUSSION THREAD

**Discussion by elyurn** [atmos-cloud-sim-uj/numba-mpi#153] [Q&A]
Hello,

I would like to know if you have a trick to avoid the JIT compilation in each process, like this for the multiprocessing package: https://stackoverflow.com/questions/63875749/avoid-recompilation-of-numba-code-when-using-multiprocessing

I must run some code on, say, more than 80 nodes and the JIT compilation takes about 2 seconds so it gets very long!

**Comment by slayoo** [comment]
Since in an MPI setup, anyhow all workers would need to wait for a one that compiles the code, the key to achieving any speedup would be to cache the compiled code across different executions, right?

It could perhaps be possible with the Numba's built-in compiled-code cache, which can be shared across different machines: https://numba.pydata.org/numba-doc/dev/developer/caching.html#cache-sharing

Never tried it, though. Numba docs furthermore mention that ["The only allowed external symbols are from the NRT or other common symbols from system libraries (i.e. libc and libm)"](https://numba.pydata.org/numba-doc/dev/developer/caching.html#requirements-for-cacheability) - it might be that our reliance on MPI lib symbols could cause lack of support for caching the compiled code.

**Reply by elyurn** [reply]
I'll give it a try when I have time, thanks.
